README.md

```diff
@@ -93,80 +93,125 @@ EncryptionParamaters = {
 
 ### API
 
-One idea a new `CredentialRequest` (et al)
-```
-dictionary CredentialElement {
-    required DOMString namespace;  // As defined in ISO 18013-5 clause 8.
-    required DOMString name;
+We propose to build upon the Credential Manager’s “identity” request, which [Federated Credential Management](https://fedidcg.github.io/FedCM/#dictdef-identitycredentialrequestoptions) is also built on top of, with the following structures:
+
+```webidl
+// The existing extension to the credential management API:
+partial dictionary CredentialRequestOptions {
+  IdentityCredentialRequestOptions identity;
 };
 
-dictionary CredentialStorageDuration {
-    // At least one of these is required.
+dictionary IdentityCredentialRequestOptions {
+  required sequence<IdentityProviderConfig> providers;
 
-    boolean forever;  // Cannot be used with any other properties.
+  // Common fields across all types of providers:
 
-    long days;  // Cannot (currently) be used with any other properties.
+  // Omitting `retention` implies that the identity information will not be stored.
+  IdentityStorageDuration retention;
 };
 
-dictionary CredentialDocumentDescriptor {
-    required DOMString documentType;  // As defined in ISO 18013-5 clause 8.
+dictionary IdentityStorageDuration {
+  // Exactly one of the following must be provided.
+  boolean forever;
+  long days;
+};
 
-    required sequence<CredentialElement> requestedElements;
+dictionary IdentityProviderConfig {
+  required string scheme;
 
-    CredentialStorageDuration desiredStorageDuration;  // Not providing this is equivalent to not asking to store.
+  // Common fields across all identity provider types:
+  required DOMString nonce;
 };
 
-dictionary CredentialDocument {
-    object data;  // The CBOR encoded `CredentialDocument` defined above.
+// extends the IdentityProviderConfig with mDoc-specific attributes
+partial dictionary IdentityProviderConfig  {
+  // For use where `scheme` is `mdoc`:
+
+  // base64-encoded reader HPKE public key.
+  required DOMString readerPublicKey;
+
+  // base64-encoded reader certification chain, if the platform requires it.
+  optional sequence<DOMString> certificates;
+
+  required DOMString documentType;
+  required sequence<MdocElement> requestedElements;
+};
+
+dictionary MdocElement {
+  required DOMString namespace;  // As defined in ISO 18013-5 clause 8.
+  required DOMString name;
 };
 
-dictionary RequestConfiguration {
-    required DOMString nonce;
+partial dictionary IdentityProviderConfig {
+  // ...
+  // Federated provider specific configs, not covered here.
+  // ...
```

Comment:
IMHO the main benefit is being able to create an ordered list of different possible sources of identity information so that the browser can construct a meaningful UI that (potentially) merges information from more than one source.